src/SourceBuild/content/eng/tools/tasks/Microsoft.DotNet.SourceBuild.Tasks.LeakDetection/MarkAndCatalogPackages.cs

```diff
@@ -113,49 +110,23 @@ public override bool Execute()
                         memStream.Seek(0, SeekOrigin.Begin);
                         catalogFileEntry.OriginalHash = sha.ComputeHash(memStream);
 
-                        // Now try to read it as an assembly
-                        memStream.Seek(0, SeekOrigin.Begin);
+                        // Add poison marker to assemblies
                         try
                         {
-                            asm = AssemblyDefinition.ReadAssembly(memStream, new ReaderParameters(ReadingMode.Deferred));
+                            AssemblyName asm = AssemblyName.GetAssemblyName(f);
```

Comment:
Nit - I realize this was existing code but the `f` variable name is not very readable.  This GitHub diff highlight how unreadable it is given the location where it is declare is not visible without further source expansion.

Reply:
Yeah, I agree. Not sure if it's worth resetting the PR and doing full local build, as CI is not going to check this.